src/main/scala/io/iohk/ethereum/keystore/KeyStore.scala

```diff
@@ -40,15 +41,34 @@ trait KeyStore {
   def changePassphrase(address: Address, oldPassphrase: String, newPassphrase: String): Either[KeyStoreError, Unit]
 }
 
-class KeyStoreImpl(keyStoreDir: String, secureRandom: SecureRandom) extends KeyStore with Logger {
+class KeyStoreImpl(keyStoreConfig: KeyStoreConfig, secureRandom: SecureRandom) extends KeyStore with Logger {
 
   init()
 
-  def newAccount(passphrase: String): Either[KeyStoreError, Address] = {
+  def newAccount(passphrase: String): Either[KeyStoreError, Address] = for {
+    _       <- validateNewPassPhrase(passphrase)
+    address <- saveNewAccount(passphrase)
+  } yield address
+
+  private def saveNewAccount(passphrase: String): Either[KeyStoreError, Address] = {
     val keyPair = generateKeyPair(secureRandom)
     val (prvKey, _) = keyPairToByteStrings(keyPair)
     val encKey = EncryptedKey(prvKey, passphrase, secureRandom)
     save(encKey).map(_ => encKey.address)
+
+  }
+
+  private def validateNewPassPhrase(passPhrase: String): Either[KeyStoreError, Unit] = {
```

Comment:
Should we also add this validation when importing pk keys? (on the `KeyStore.importPrivateKey` method)

Reply:
I was thinking about it, but my conclusion was that it is better to allow to import already created key with an even too short passphrase  (therefore not scare user who wants to try mantis). All trouble he will have later is to change this passphrase (he will need to change it to empty and then to what he wants or mingle with configs.) wdyt?

Reply:
as per discussion corrected :)